atlassian_jwt_auth/key.py

```diff
@@ -93,7 +93,10 @@ def retrieve(self, key_identifier, **requests_kwargs):
             key_identifier = KeyIdentifier(key_identifier)
 
         url = self.base_url + key_identifier.key_id
-        return self._retrieve(url, requests_kwargs)
```

Comment:
I prefer not to wrap exceptions like this unless really needed.

Reply:
That would mean that any code that cares about handling failures to retrieve public keys must know all the details about how those keys are received, which is a leaky abstraction.

I'm fine with a more elaborate wrapping scheme than what's currently here, e.g. adding an `original_exception` attribute to `PublicKeyRetrieverException`.

Reply:
I agree with Grant. This code here should not be coupled to the fact that there is a aiohttp retriever.
Currently the retriever interface exposes `requests` and that's what clients should anticipate. While we *could* abstract the `RequestException` with something implementation-agnostic, I think it wouldn't add too much value.